slsReceiverSoftware/src/BinaryDataFile.cpp

```diff
@@ -4,88 +4,92 @@
 
 namespace sls {
 
-BinaryDataFile::BinaryDataFile(const int index) : File(BINARY), index_(index) {}
+BinaryDataFile::BinaryDataFile(const int index) : index(index) {}
 
 BinaryDataFile::~BinaryDataFile() { CloseFile(); }
 
+slsDetectorDefs::fileFormat BinaryDataFile::GetFileFormat() const {
+    return BINARY;
+}
+
 void BinaryDataFile::CloseFile() {
-    if (fd_) {
-        fclose(fd_);
+    if (fd) {
+        fclose(fd);
     }
-    fd_ = nullptr;
+    fd = nullptr;
 }
 
 void BinaryDataFile::CreateFirstBinaryDataFile(
-    const std::string filePath, const std::string fileNamePrefix,
-    const uint64_t fileIndex, const bool overWriteEnable, const bool silentMode,
-    const int modulePos, const int numUnitsPerReadout,
-    const uint32_t udpPortNumber, const uint32_t maxFramesPerFile) {
-
-    subFileIndex_ = 0;
-    numFramesInFile_ = 0;
-
-    filePath_ = filePath;
-    fileNamePrefix_ = fileNamePrefix;
-    fileIndex_ = fileIndex;
-    overWriteEnable_ = overWriteEnable;
-    silentMode_ = silentMode;
-    detIndex_ = modulePos;
-    numUnitsPerReadout_ = numUnitsPerReadout;
-    udpPortNumber_ = udpPortNumber;
-    maxFramesPerFile_ = maxFramesPerFile;
+    const std::string fPath, const std::string fNamePrefix,
+    const uint64_t fIndex, const bool ovEnable, const bool sMode,
+    const int modulePos, const int nUnitsPerReadout,
+    const uint32_t uPortNumber, const uint32_t mFramesPerFile) {
+
+    subFileIndex = 0;
+    numFramesInFile = 0;
+
+    filePath = fPath;
+    fileNamePrefix = fNamePrefix;
```

Comment:
Do we need both fileNamePrefix and fNamePrefix? It looks like fNamePrefix is just used to set fileNamePrefix...

Same question applies to the other members.

Reply:
one is a data member and the other is an actual argument used to set it upon the first file creation. For consecutive file creations in the same acquisition, the data member is then used.

Reply:
My bad looked quickly and mistook it for the constructor. I wonder if we could pass these arguments to the constructor and then do the creation by calling CreateFille when needed. But that might mess up the other structure...

Reply:
Well, I am hoping in the next PR to use the set methods to set up these members when the class objects are created and when these parameters are changed from the client. So we would be throwing away quite a few parameters. Also, planning to set one value instead of  fname, detindex and uniterperreadout as this string will not change for a particular thread.